lib/explorer/backend/series.ex

```diff
@@ -23,6 +23,7 @@ defmodule Explorer.Backend.Series do
   @callback to_iovec(s) :: [binary()]
   @callback cast(s, dtype) :: s
   @callback categorise(s, s) :: s
+  @callback strptime(s, String.t() | nil) :: s
```

Comment:
If you want to cast a string to datetime (i.e. nil), you call `cast(:datetime)`. So I think we can remove `nil` and we document that `cast(:date)` and `cast(:datetime)` are also options. :)